### PR TITLE
[20.0][Create Timepoint] Actually check visit label in validation function

### PR DESCRIPTION
**Pull request by ZainVirani**
This PR addresses [Redmine #14782](https://redmine.cbrain.mcgill.ca/issues/14782).
Specifically, if a user selects a subproject and visit label, and then switches the subproject to one where its no longer valid and clicks 'Create Timepoint' quickly enough, they can create a timepoint with an invalid visit label (this issue would also be solved by reactifying this module, but we can't do that before 20.0 is released).

I added a check to `isValidVisitLabel()` in `TimePoint.class.inc` to see if the submitted Visit Label is assigned to the submitted subproject by checking the config setting (XML).

To test this PR follow the steps outlined in the redmine ticket.

### CODE REVIEW

**Comment by PapillonMcGill** [comment]
This could be done using the new Visit class. I believe it as been included in the major (20.0) version.  Next step for the Visit class, is to convert all the usage of VisitLabel (in any form) to use this class. 

@driusan could you confirm in which version Visit class is added?

File: php/libraries/TimePoint.class.inc [hunk 1]
```diff
@@ -272,6 +272,27 @@ class TimePoint
...
+        $visitLabelSettings = $config->getSetting('visitLabel');
+        $isValid            = false;
+        foreach (\Utility::toArray($visitLabelSettings) as $visitLabelSetting) {
+            if ($visitLabelSetting['@']['subprojectID'] == $subprojectID) {
+                $items = \Utility::toArray(
+                    $visitLabelSetting['labelSet']['item']
+                );
+                foreach ($items as $item) {
+                    if ($item['#'] == $visitLabel) {
+                        $isValid = true;
```
Comment (davidblader) [inline php/libraries/TimePoint.class.inc]: maybe add a break statement(s) once you know it's valid

Comment (ZainVirani) [reply php/libraries/TimePoint.class.inc]: good point

Comment (ZainVirani) [reply php/libraries/TimePoint.class.inc]: @davidblader done

**Comment by ZainVirani** [comment]
@davidblader please re review

**Pull request merged**